cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java

```diff
@@ -0,0 +1,219 @@
+
+import com.hp.oo.sdk.content.annotations.Param;
+import io.cloudslang.lang.entities.AsyncLoopStatement;
+import io.cloudslang.lang.entities.ResultNavigation;
+import io.cloudslang.lang.entities.ScoreLangConstants;
+import io.cloudslang.lang.entities.bindings.Output;
+import io.cloudslang.lang.runtime.bindings.AsyncLoopBinding;
+import io.cloudslang.lang.runtime.bindings.LoopsBinding;
+import io.cloudslang.lang.runtime.bindings.OutputsBinding;
+import io.cloudslang.lang.runtime.env.*;
+import io.cloudslang.lang.runtime.events.LanguageEventData;
+import org.apache.commons.lang.SerializationUtils;
+import org.apache.log4j.Logger;
+import io.cloudslang.score.api.EndBranchDataContainer;
+import io.cloudslang.score.api.execution.ExecutionParametersConsts;
+import io.cloudslang.score.lang.ExecutionRuntimeServices;
+import org.python.google.common.collect.Lists;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.cloudslang.score.api.execution.ExecutionParametersConsts.EXECUTION_RUNTIME_SERVICES;
+
+/**
+ * Date: 3/25/2015
+ *
+ * @author Bonczidai Levente
+ */
+@Component
+public class AsyncLoopSteps extends AbstractSteps {
+
+    @Autowired
+    private AsyncLoopBinding asyncLoopBinding;
+
+    @Autowired
+    private LoopsBinding loopsBinding;
+
+    @Autowired
+    private OutputsBinding outputsBinding;
+
+    private static final Logger logger = Logger.getLogger(AsyncLoopSteps.class);
+
+    public void addBranches(@Param(ScoreLangConstants.ASYNC_LOOP_STATEMENT_KEY) AsyncLoopStatement asyncLoopStatement,
+                            @Param(ScoreLangConstants.RUN_ENV) RunEnvironment runEnv,
+                            @Param(EXECUTION_RUNTIME_SERVICES) ExecutionRuntimeServices executionRuntimeServices,
+                            @Param(ScoreLangConstants.NODE_NAME_KEY) String nodeName,
+                            @Param(ExecutionParametersConsts.RUNNING_EXECUTION_PLAN_ID) Long RUNNING_EXECUTION_PLAN_ID,
+                            @Param(ScoreLangConstants.NEXT_STEP_ID_KEY) Long nextStepId,
+                            @Param(ScoreLangConstants.BRANCH_BEGIN_STEP_ID_KEY) Long branchBeginStep,
+                            @Param(ScoreLangConstants.REF_ID) String refId) {
+
+        try {
+            Context flowContext = runEnv.getStack().popContext();
+
+            List<Serializable> splitData = asyncLoopBinding.bindAsyncLoopList(asyncLoopStatement, flowContext, nodeName);
+
+            fireEvent(
+                    executionRuntimeServices,
+                    runEnv,
+                    ScoreLangConstants.EVENT_ASYNC_LOOP_INPUT_END,
+                    "async loop list evaluated",
+                    Pair.of(LanguageEventData.BOUND_ASYNC_LOOP_INPUTS, (Serializable) splitData),
```

Comment:
I meant here that the split-list is evaluated in the addBranches (split) step: e.g. `for: value in str.split()` -> [1,2,3].. what do you think would be a better name?